File: Data-Structure/Nodejs/Top Interview Questions/Array/Easy/217. Contains Duplicate.js
//217. Contains Duplicate
/*

**** read carefully **** 
Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.

Input: nums = [1,2,3,1]
Output: true

Input: nums = [1,2,3,4]
Output: false
 
Input: nums = [1,1,1,3,3,4,3,2,4,2]
Output: true
*/
//1 : sort T: O(NlogN); S: O(1)
//2 : hash Set
{
    //T: O(N); 
    //S: O(N)
    var containsDuplicate = function (nums) {
        let set = new Set(nums);
        return set.size !== nums.length;
    }
}

{
    // give arr, from num 1 ~ n has duplicate
}

File: Data-Structure/_All-In-One-LeetCode/00217. Contains Duplicate/217. Contains Duplicate.cs
internal class Program
{
    private static void Main(string[] args)
    {
        int[] nums = { 1, 2, 3, 4 };
        containsDuplicate(nums);
    }

    public boolean containsDuplicate(int[] nums)
    {

        Set<Integer> distinct = new HashSet<Integer>();
        foreach (int num in nums)
        {
            if (distinct.contains(num))
            {
                return true;
            }
            distinct.add(num);
        }
        return false;
    }

    private static boolean containsDuplicate1(int[] nums)
    {

        for (int i = 0; i < nums.length; i++)
        {
            for (int j = i + 1; j < nums.length; j++)
            {
                if (nums[i] == nums[j])
                {
                    return true;
                }
            }
        }
        return false;
    }

    public boolean containsDuplicate2(int[] nums)
    {

        Arrays.sort(nums);
        for (int ind = 1; ind < nums.length; ind++)
        {
            if (nums[ind] == nums[ind - 1])
            {
                return true;
            }
        }
        return false;
    }
}